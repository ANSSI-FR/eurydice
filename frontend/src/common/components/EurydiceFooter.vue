<template>
  <v-footer padless>
    <v-row justify="center" no-gutters>
      <v-btn href="/api/docs/" text class="my-2">
        Documentation de l'API
      </v-btn>
      <v-dialog v-model="dialogVisible" width="500">
        <template #activator="{ on, attrs }">
          <v-btn v-bind="attrs" text class="my-2" v-on="on">Contact</v-btn>
        </template>
        <v-card>
          <v-card-title>Contact</v-card-title>
          <v-card-text class="align-center">
            En cas de probl√®me ou pour nous faire part de vos retours, veuillez
            nous contacter {{ appContact }}.
          </v-card-text>
          <v-card-actions>
            <v-spacer />
            <v-btn color="primary" text @click="dialogVisible = false"
              >Fermer</v-btn
            >
          </v-card-actions>
        </v-card>
      </v-dialog>
    </v-row>
  </v-footer>
</template>

<script>
import getServerConfig from "@common/api/serverMetadata";
import { mapGetters } from "vuex";

export default {
  name: "EurydiceFooter",
  data() {
    return {
      dialogVisible: false,
    };
  },
  computed: {
    ...mapGetters(["serverMetadata"]),
    appContact() {
      return this.serverMetadata?.contact;
    },
    badgeContent() {
      return this.serverMetadata?.badgeContent;
    },
  },
  async beforeCreate() {
    // TODO this should be done by a higher-level component
    const serverMetadata = await getServerConfig();
    this.$store.commit("setServerMetadata", serverMetadata);
  },
};
</script>
