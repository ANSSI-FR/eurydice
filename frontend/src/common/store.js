import Vue from "vue";
import Vuex from "vuex";

Vue.use(Vuex);

export default new Vuex.Store({
  state: () => ({
    username: null,
    maintenance: false,
    serverDown: false,
    serverMetadata: null,
    ongoingTransfer: false,
  }),
  getters: {
    maintenance(state) {
      return state.maintenance;
    },
    serverDown(state) {
      return state.serverDown;
    },
    serverMetadata(state) {
      return state.serverMetadata;
    },
    ongoingTransfer(state) {
      return state.ongoingTransfer;
    },
  },
  mutations: {
    setUsername(state, newUsername) {
      state.username = newUsername;
    },
    setMaintenance(state, newMaintenance) {
      state.maintenance = newMaintenance;
    },
    setServerDown(state, serverDown) {
      state.serverDown = serverDown;
    },
    setServerMetadata(state, serverMetadata) {
      state.serverMetadata = serverMetadata;
    },
    setOngoingTransfer(state, ongoingTransfer) {
      state.ongoingTransfer = ongoingTransfer;
    },
  },
  actions: {},
  // NOTE: consider adding modules to this common store in main.js
  //       when adding origin/destination specific state/mutations
  modules: {},
});
