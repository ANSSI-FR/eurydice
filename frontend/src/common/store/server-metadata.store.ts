import * as serverMetadataService from '@common/services/server-metadata.service';
import { defineStore } from 'pinia';
import { computed, ref } from 'vue';

export const useServerMetadataStore = defineStore('serverMetadataStore', () => {
  const serverMetadata = ref<ServerMetadata>();

  const getServerMetadata = computed(() => serverMetadata.value);
  const setServerMetadata = (metadata: ServerMetadata | undefined) => {
    serverMetadata.value = metadata;
  };
  const fetchServerMetadata = async () => {
    await serverMetadataService.getServerMetadata().then((fetchedServerMetadata) => {
      setServerMetadata(fetchedServerMetadata);
    });
  };
  const $reset = () => {
    serverMetadata.value = undefined;
  };
  return {
    getServerMetadata,
    setServerMetadata,
    fetchServerMetadata,
    $reset,
  };
});
