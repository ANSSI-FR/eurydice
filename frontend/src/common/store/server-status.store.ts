import { defineStore } from 'pinia';
import { computed, ref } from 'vue';

export const useServerStatusStore = defineStore('serverStatusStore', () => {
  const isServerDown = ref<boolean>(false);
  const isServerInMaintenance = ref<boolean>(false);

  const getIsServerDown = computed(() => isServerDown.value);
  const setIsServerDown = (value: boolean) => {
    isServerDown.value = value;
  };

  const getIsServerInMaintenance = computed(() => isServerInMaintenance.value);
  const setIsServerInMaintenance = (value: boolean) => {
    isServerInMaintenance.value = value;
  };

  const $reset = () => {
    isServerDown.value = false;
    isServerInMaintenance.value = false;
  };
  return {
    getIsServerDown,
    setIsServerDown,
    getIsServerInMaintenance,
    setIsServerInMaintenance,
    $reset,
  };
});
