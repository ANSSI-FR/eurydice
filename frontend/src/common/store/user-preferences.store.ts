import type { UserPreferences } from '@common/models/UserPreferences';
import { defineStore } from 'pinia';
import { readonly, ref } from 'vue';

export const useUserPreferencesStore = defineStore('userPreferencesStore', () => {
  const LOCAL_STORAGE_KEY: string = 'userPreferences';

  const updateDarkModeInHTML = (darkMode: boolean | undefined) => {
    if (darkMode) {
      document.querySelector('html')!.classList.add('dark');
    } else {
      document.querySelector('html')!.classList.remove('dark');
    }
  };

  const initFromLocalStorage = () => {
    let preferences;
    const localStoragePreferences = localStorage.getItem(LOCAL_STORAGE_KEY);
    if (localStoragePreferences) {
      preferences = JSON.parse(localStoragePreferences);
    } else {
      const systemDarkMode = window.matchMedia('(prefers-color-scheme: dark)').matches;
      preferences = { darkMode: systemDarkMode };
    }
    updateDarkModeInHTML(preferences.darkMode);
    return preferences;
  };

  const saveUserPreferencesToLocalStorage = () => {
    localStorage.setItem(LOCAL_STORAGE_KEY, JSON.stringify(userPreferences.value));
  };

  const userPreferences = ref<UserPreferences | undefined>(initFromLocalStorage());

  const getUserPreferences = readonly(userPreferences);

  const setDarkModePreference = (darkMode: boolean) => {
    if (!userPreferences.value) {
      userPreferences.value = {};
    }
    if (userPreferences.value.darkMode !== darkMode) {
      userPreferences.value.darkMode = darkMode;
      saveUserPreferencesToLocalStorage();
      updateDarkModeInHTML(darkMode);
    }
  };

  return {
    getUserPreferences,
    setDarkModePreference,
  };
});
