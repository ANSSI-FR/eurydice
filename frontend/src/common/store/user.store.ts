import { defineStore } from 'pinia';
import { computed, ref } from 'vue';

export const useUserStore = defineStore('userStore', () => {
  const currentUser = ref<User>();

  const getCurrentUser = computed(() => currentUser.value);
  const setCurrentUser = (user: User | undefined) => {
    currentUser.value = user;
  };
  const $reset = () => {
    currentUser.value = undefined;
  };
  return {
    getCurrentUser,
    setCurrentUser,
    $reset,
  };
});
