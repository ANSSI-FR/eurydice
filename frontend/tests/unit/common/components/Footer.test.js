import EurydiceFooter from "@common/components/EurydiceFooter";
import { mount } from "@vue/test-utils";
import store from "@common/store";

describe("Mounted EurydiceFooter", () => {
  const wrapper = mount(EurydiceFooter, { store });

  it("renders correctly", async () => {
    await wrapper.setData({
      links: [
        { label: "foo", url: "bar" },
        {
          label: "i like",
          url: "waffles",
        },
      ],
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
