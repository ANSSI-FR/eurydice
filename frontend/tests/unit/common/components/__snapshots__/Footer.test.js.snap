// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mounted EurydiceFooter renders correctly 1`] = `
<footer
  class="v-footer v-sheet theme--light v-footer--padless"
>
  <div
    class="row no-gutters justify-center"
  >
    <a
      class="my-2 v-btn v-btn--text theme--light v-size--default"
      href="/api/docs/"
    >
      <span
        class="v-btn__content"
      >
        
      Documentation de l'API
    
      </span>
    </a>
     
    <div
      class="v-dialog__container"
    >
      <!---->
    </div>
    <button
      aria-expanded="false"
      aria-haspopup="true"
      class="my-2 v-btn v-btn--text theme--light v-size--default"
      role="button"
      type="button"
    >
      <span
        class="v-btn__content"
      >
        Contact
      </span>
    </button>
  </div>
</footer>
`;
